src/engraving/libmscore/cmd.cpp

```diff
@@ -3210,7 +3210,7 @@ void Score::cmdExplode()
                     // keep note "i" from top, which is backwards from nnotes - 1
                     // reuse notes if there are more instruments than notes
                     size_t stavesPerNote = std::max((lastStaff - srcStaff) / nnotes, static_cast<size_t>(1));
-                    size_t keepIndex = std::max(nnotes - 1 - (i / stavesPerNote), static_cast<size_t>(0));
+                    size_t keepIndex = static_cast<size_t>(std::max(static_cast<int>(nnotes) - 1 - static_cast<int>(i / stavesPerNote), 0));
```

Comment:
Too many static_cast... What exactly went wrong with this expression?

Reply:
`nnotes - 1 - (i / stavesPerNote)` is `size_t`, which means that doing a subtraction and calling `std::max(x, 0)` doesn't help when x underflows to `size_t::MAX`. This was the cause of the crash. So I cast that as a signed integer so we can get negative numbers, and then cast the thing back to `size_t` when we are guaranteed a minimum value of zero.

Reply:
I see, now I understand the problem :) But shouldn't a similar change be applied to the line above?

Reply:
`lastStaff` is never less than `srcStaff` (because of the `for` loop restrictions) and `nnotes` is unsigned, so underflow isn't going to happen on that line I think